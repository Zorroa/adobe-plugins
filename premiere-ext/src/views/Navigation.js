import React, { useState } from "react"
import SearchButton from '../comp/SearchButton'
import SettingsButton from '../comp/SettingsButton'
import { useHistory } from "react-router-dom"
function Navigation(props) {
    const history = useHistory()

    const [term, setTerm] = useState("")
    const [type, setType] = useState("all")

    const onTypeChanged = (evt)=>{
        setType(evt.target.value)
    }
    const onSearch = (evt) => {
        history.push("/search/" + term + "/" + type)
      }
    
      const onSettings = (evt) => {
        history.push("/settings")
      }
    return (
        <div>
            <div className="row">
                <div className="col-3">
                    <svg className="logo-padding" viewBox="0 0 2265 673" fill="#fff" height="30"><path d="M238.3.8l270.2 93.7-15.2 37.8zm-5.8 20.5l286.1 145.3-150.4 290.7-57.4-32.7 165.5-222.2z"></path><path d="M197 21.3l251 190.8-206.3 256.2-42.3-35.1L406.8 236z"></path><path d="M300.5 207.2L22.6 274.3 33 312.7zM300 231L.6 346.2l121 305L183 623 39.7 385z"></path><path d="M333.8 231L67.4 398.8 248 672.3l46-30.8-188-215.8zm1841.8-2H2131l-88.9 223.5h46.2l22.2-54.3h85.5l22.2 54.6h46.2l-89-223.7zm6.8 129.8h-58.1l29-88.8 29.1 88.8zM1679.3 333a59.6 59.6 0 0013.7-37.5 58 58 0 00-29-51.2 114 114 0 00-58.2-13.6h-97.5V454h42.8v-92h59.8l46.2 92.3h44.4l-49.4-100.7c10.4-5 19.7-12 27.2-20.6zm-29-39.2a31 31 0 01-10.3 23.9c-4.8 3.5-10 6.3-15.4 8.5-7.3 1.5-14.8 2-22.2 1.7h-52.9v-64.7h56.4c10.5-.6 21 1.2 30.8 5.2 8.9 5.4 14 15.2 13.6 25.6v-.2zm-248 39.2a59.4 59.4 0 0013.7-37.5 58 58 0 00-29-51.2 113.9 113.9 0 00-58.2-13.6h-97.4V454h42.7v-92h59.9l46.1 92.3h44.5L1375 353.7a82 82 0 0027.3-20.6zm-29-39c.2 9-3.5 17.8-10.3 23.8-4.7 3.5-9.9 6.4-15.4 8.6-7.3 1.4-14.7 2-22.2 1.7h-53v-64.9h56.4c10.6-.6 21 1.2 30.8 5.2 9 5.3 14.2 15.2 13.7 25.6zM734.6 416.6L861 257.2v-28.5H681.8v38.5h119.5l-124 153.6-.6.6V455h187.6v-38.4zM1081.2 233c-90.3-31.4-178.3 56.4-146.7 146.6a115.9 115.9 0 0071 70.8c90.2 31.5 178.2-56.4 146.7-146.5a116 116 0 00-71-70.9zm35.7 89.8a76.3 76.3 0 01-92.6 92.4 77.1 77.1 0 01-54.6-54.2 76.3 76.3 0 0192.7-92.5 77.3 77.3 0 0154.5 54.3zm803.7-89.8c-90.2-31.3-178.2 56.4-146.8 146.5a116.2 116.2 0 0071 70.9c90.3 31.6 178.4-56.4 146.8-146.5a116 116 0 00-71-70.9zm34 89.8a76.3 76.3 0 01-92.6 92.4 77.1 77.1 0 01-54.5-54.5 76.3 76.3 0 0192.6-92.4 77.1 77.1 0 0154.6 54.5z"></path></svg>
                </div>
                <div className="col-6">
                    <div className="row"><div className="col-12">
                        <input type="text" placeholder="search term" onChange={e => setTerm(e.target.value)}/>
                        <SearchButton onpress={onSearch} /></div></div>
                    <div className="row no-padding">
                        <div className="col-12">
                            <ul><li><input type="radio" name="type" value="all" checked={type === "all"} onChange={onTypeChanged} /><label className="ul-label" htmlFor="all">all</label></li>
                                <li><input type="radio" name="type" value="video" checked={type === "video"}  onChange={onTypeChanged}/><label className="ul-label" htmlFor="video">video</label></li>
                                <li><input type="radio" name="type" value="image" checked={type === "image"}  onChange={onTypeChanged}/><label className="ul-label" htmlFor="image">image</label></li>
                                <li><input type="radio" name="type" value="document" checked={type === "document"}  onChange={onTypeChanged}/><label className="ul-label" htmlFor="document">document</label></li>
                            </ul>
                        </div>
                    </div>
                </div>
                <div className="col-3 text-right logo-padding"><SettingsButton  onpress={onSettings} /></div>
            </div>
        </div>
    )
}
export default Navigation