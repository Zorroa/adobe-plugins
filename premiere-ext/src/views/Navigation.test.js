import renderer from "react-test-renderer"
import Navigation from "./Navigation"
import {
  Form,
  FormControl,
  Button,
  InputGroup,
  DropdownButton,
  Dropdown,
} from "react-bootstrap"

describe("<Navigation/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Navigation />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
