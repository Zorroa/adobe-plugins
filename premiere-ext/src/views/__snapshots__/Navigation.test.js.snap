// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <svg
        className="logo-padding"
        fill="#fff"
        height="30"
        viewBox="0 0 2265 673"
      >
        <path
          d="M238.3.8l270.2 93.7-15.2 37.8zm-5.8 20.5l286.1 145.3-150.4 290.7-57.4-32.7 165.5-222.2z"
        />
        <path
          d="M197 21.3l251 190.8-206.3 256.2-42.3-35.1L406.8 236z"
        />
        <path
          d="M300.5 207.2L22.6 274.3 33 312.7zM300 231L.6 346.2l121 305L183 623 39.7 385z"
        />
        <path
          d="M333.8 231L67.4 398.8 248 672.3l46-30.8-188-215.8zm1841.8-2H2131l-88.9 223.5h46.2l22.2-54.3h85.5l22.2 54.6h46.2l-89-223.7zm6.8 129.8h-58.1l29-88.8 29.1 88.8zM1679.3 333a59.6 59.6 0 0013.7-37.5 58 58 0 00-29-51.2 114 114 0 00-58.2-13.6h-97.5V454h42.8v-92h59.8l46.2 92.3h44.4l-49.4-100.7c10.4-5 19.7-12 27.2-20.6zm-29-39.2a31 31 0 01-10.3 23.9c-4.8 3.5-10 6.3-15.4 8.5-7.3 1.5-14.8 2-22.2 1.7h-52.9v-64.7h56.4c10.5-.6 21 1.2 30.8 5.2 8.9 5.4 14 15.2 13.6 25.6v-.2zm-248 39.2a59.4 59.4 0 0013.7-37.5 58 58 0 00-29-51.2 113.9 113.9 0 00-58.2-13.6h-97.4V454h42.7v-92h59.9l46.1 92.3h44.5L1375 353.7a82 82 0 0027.3-20.6zm-29-39c.2 9-3.5 17.8-10.3 23.8-4.7 3.5-9.9 6.4-15.4 8.6-7.3 1.4-14.7 2-22.2 1.7h-53v-64.9h56.4c10.6-.6 21 1.2 30.8 5.2 9 5.3 14.2 15.2 13.7 25.6zM734.6 416.6L861 257.2v-28.5H681.8v38.5h119.5l-124 153.6-.6.6V455h187.6v-38.4zM1081.2 233c-90.3-31.4-178.3 56.4-146.7 146.6a115.9 115.9 0 0071 70.8c90.2 31.5 178.2-56.4 146.7-146.5a116 116 0 00-71-70.9zm35.7 89.8a76.3 76.3 0 01-92.6 92.4 77.1 77.1 0 01-54.6-54.2 76.3 76.3 0 0192.7-92.5 77.3 77.3 0 0154.5 54.3zm803.7-89.8c-90.2-31.3-178.2 56.4-146.8 146.5a116.2 116.2 0 0071 70.9c90.3 31.6 178.4-56.4 146.8-146.5a116 116 0 00-71-70.9zm34 89.8a76.3 76.3 0 01-92.6 92.4 77.1 77.1 0 01-54.5-54.5 76.3 76.3 0 0192.6-92.4 77.1 77.1 0 0154.6 54.5z"
        />
      </svg>
    </div>
    <div
      className="col-6"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <input
            onChange={[Function]}
            placeholder="search term"
            type="text"
          />
          <button
            className="btn btn-sm z-button btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              id="settingsIcon"
            >
              <svg
                className="bi bi-search"
                fill="currentColor"
                height="1em"
                viewBox="0 0 18 18"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.442 10.442a1 1 0 0 1 1.415 0l3.85 3.85a1 1 0 0 1-1.414 1.415l-3.85-3.85a1 1 0 0 1 0-1.415z"
                  fillRule="evenodd"
                />
                <path
                  d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zM13 6.5a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0z"
                  fillRule="evenodd"
                />
              </svg>
               
              Search
            </span>
          </button>
        </div>
      </div>
      <div
        className="row no-padding"
      >
        <div
          className="col-12"
        >
          <ul>
            <li>
              <input
                checked={true}
                name="type"
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <label
                className="ul-label"
                htmlFor="all"
              >
                all
              </label>
            </li>
            <li>
              <input
                checked={false}
                name="type"
                onChange={[Function]}
                type="radio"
                value="video"
              />
              <label
                className="ul-label"
                htmlFor="video"
              >
                video
              </label>
            </li>
            <li>
              <input
                checked={false}
                name="type"
                onChange={[Function]}
                type="radio"
                value="image"
              />
              <label
                className="ul-label"
                htmlFor="image"
              >
                image
              </label>
            </li>
            <li>
              <input
                checked={false}
                name="type"
                onChange={[Function]}
                type="radio"
                value="document"
              />
              <label
                className="ul-label"
                htmlFor="document"
              >
                document
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-3 text-right logo-padding"
    >
      <button
        className="btn btn-default btn-sm  btn btn-primary"
        disabled={false}
        id="settingsButton"
        onClick={[Function]}
        type="button"
      >
        <span
          id="settingsIcon"
        >
          <svg
            className="bi bi-gear-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 18 18"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 0 0-5.86 2.929 2.929 0 0 0 0 5.858z"
              fillRule="evenodd"
            />
          </svg>
           
          Settings
        </span>
      </button>
    </div>
  </div>
</div>
`;
